### PR TITLE
log: log level per channel

### PR DESCRIPTION
**Pull request by yicooo**
Commit `e1d72fb62092631738b376fea846c4d6f4caf3dc` adds log level filter per channel for iowriter log handler. 

```yaml
    handlers:
        main:
            type: iowriter
            channels:
              # Excludes logs from "metrics" channel altogether.
              - name: metrics
                disabled: false
              # Logs debug and above for "test" channel.
              - name: test
                level: debug
            formatter: console
            level: info
            timestamp_format: 15:04:05.000
            writer: stdout
```

Commit `c0b05bf2e44925182c85a60022e2c0f986cd25e4` fixes test failures, mostly due to package updates. Can be removed from this PR depending on the needs.

### CODE REVIEW

File: pkg/apiserver/crud/crud_test.go [hunk 1]
```diff
@@ -7,15 +7,16 @@ import (
 	"testing"
 	"time"
 
+	"github.com/stretchr/testify/assert"
```
Comment (netsi) [inline pkg/apiserver/crud/crud_test.go]: I think you might have different import formatting settings than the ones used in this repo

File: pkg/kvstore/ddb_test.go [hunk 2]
```diff
@@ -4,12 +4,13 @@ import (
 	"context"
 	"testing"
 
+	"github.com/stretchr/testify/assert"
```
Comment (netsi) [inline pkg/kvstore/ddb_test.go]: here too, this doesn't match the import formatting standards of the proejct

File: pkg/kvstore/redis_test.go [hunk 3]
```diff
@@ -6,12 +6,13 @@ import (
 	"testing"
 	"time"
 
+	"github.com/stretchr/testify/assert"
```
Comment (netsi) [inline pkg/kvstore/redis_test.go]: also here

File: pkg/lambda/lambda.go [hunk 4]
```diff
@@ -5,6 +5,7 @@ import (
 	"os"
 
 	awsLambda "github.com/aws/aws-lambda-go/lambda"
+
```
Comment (netsi) [inline pkg/lambda/lambda.go]: please fix the import formatting settings

File: pkg/log/logger.go [hunk 5]
```diff
@@ -178,13 +177,7 @@ func (l *gosoLogger) log(level int, msg string, args []interface{}, loggedErr er
...
 
 	for _, handler := range l.handlers {
-		if handler.Level() > level {
-			continue
-		}
-
-		handlerChannels := handler.Channels()
-
-		if !l.shouldLogToChannels(l.data.Channel, handlerChannels) {
+		if !l.shouldLog(l.data.Channel, level, handler) {
```
Comment (netsi) [inline pkg/log/logger.go:180]: seems like we only use it here, maybe we should instead have a function shouldNotLog

Comment (yicooo) [reply pkg/log/logger.go:180]: ignoring, this is a common usage

**Review by netsi — APPROVED** [review]

**Review by netsi — APPROVED** [review]

**Pull request merged**